proseco/guide.py

```diff
@@ -814,3 +858,55 @@ def spoiled_by_bad_pixel(cand_guides, dither):
                         'text': (f'Cand {cand["id"]} spoiled by {np.count_nonzero(bps)} bad pixels '
                                  f'including {(bp_row[bps][0], bp_col[bps][0])}')})
     return spoiled, rej
+
+
+def dist2(g1, g2):
+    """
+    Calculate squared distance between a pair of stars in a star table.
+
+    This local version of the method uses a cache.
+    """
+    if (g1['id'], g2['id']) in STAR_PAIR_DIST_CACHE:
+        return STAR_PAIR_DIST_CACHE[(g1['id'], g2['id'])]
+    out = (g1['yang'] - g2['yang']) ** 2 + (g1['zang'] - g2['zang']) ** 2
+    STAR_PAIR_DIST_CACHE[(g1['id'], g2['id'])] = out
+    return out
+
+
+def check_single_cluster(cand_guide_set, threshold, n_minus):
+    """
+    Confirm that a set of stars satisfies the minimum separation threshold when n_minus
+    stars are removed from the set.  For example, for a threshold of 1000, n_minus of 1,
+    and an input set of candidates stars with 5 candidates, confirm that, for any 4 of
+    the 5 stars (5 minus n_minus = 1), one pair of those stars is separated by at
+    least 1000 arcsecs.
+
+    :rtype: bool
```

Comment:
Docs.

Reply:
Not sure if I address the Docs issue here.